src/ui/reducers/network.ts

```diff
@@ -1,31 +1,52 @@
-import { createSelector } from "reselect";
 import { RequestEventInfo, RequestInfo } from "@recordreplay/protocol";
 import { UIState } from "ui/state";
 import { NetworkAction } from "ui/actions/network";
+import { WiredFrame } from "protocol/thread/pause";
+import { createSelector } from "reselect";
+import { getSources } from "devtools/client/debugger/src/reducers/sources";
+import { formatCallStackFrames } from "devtools/client/debugger/src/selectors/getCallStackFrames";
 
 export type NetworkState = {
   events: RequestEventInfo[];
+  frames: Record<string, WiredFrame[]>;
   requests: RequestInfo[];
 };
 
 const initialState = (): NetworkState => ({
   events: [],
+  frames: {},
   requests: [],
 });
 
 const update = (state: NetworkState = initialState(), action: NetworkAction): NetworkState => {
   switch (action.type) {
     case "NEW_NETWORK_REQUESTS":
       return {
+        ...state,
         events: [...action.payload.events, ...state.events],
         requests: [...action.payload.requests, ...state.requests],
       };
+    case "SET_FRAMES":
+      return {
+        ...state,
+        frames: {
+          ...state.frames,
+          [action.payload.point]: action.payload.frames,
```

Comment:
no longer sure if we need to cache here... probably fine if it is just 

```
return { ...state, frames: action.frames }
```

The one caveat is we'd have to clear it when a new row is selected. Dont feel strongly

Reply:
It can work either way, but I'm not a big fan of implicit IDs. When storing state in redux, I think a structure like:

```
{ currentFrames: [] }
```

is *far* inferior to:

{
  currentRequestId: "123",
  framesByRequestId: {
    123: []
  }
}
```

For me, it's not about caching, it's about clarity. Looking at the problems we're dealing with on the billing side right now, it's just another version of this exact problem. I am of the opinion that local data stores should imitate the structure of the source that they are pulling from, unless there is a very good reason not to.

OK, end of rant 😄